gitlab/client.py

```diff
@@ -793,51 +849,57 @@ def _query(self, url, query_data=None, **kwargs):
         self._current = 0
 
     @property
-    def current_page(self):
+    def current_page(self) -> int:
         """The current page number."""
+        assert self._current_page is not None
         return int(self._current_page)
 
     @property
-    def prev_page(self):
+    def prev_page(self) -> Optional[int]:
         """The previous page number.
 
         If None, the current page is the first.
         """
         return int(self._prev_page) if self._prev_page else None
 
     @property
-    def next_page(self):
+    def next_page(self) -> Optional[int]:
         """The next page number.
 
         If None, the current page is the last.
         """
         return int(self._next_page) if self._next_page else None
 
     @property
-    def per_page(self):
+    def per_page(self) -> int:
         """The number of items per page."""
+        assert self._per_page is not None
         return int(self._per_page)
 
     @property
-    def total_pages(self):
+    def total_pages(self) -> int:
         """The total number of pages."""
+        assert self._total_pages is not None
         return int(self._total_pages)
 
     @property
-    def total(self):
+    def total(self) -> int:
         """The total number of items."""
+        assert self._total is not None
         return int(self._total)
 
-    def __iter__(self):
+    def __iter__(self) -> "GitlabList":
         return self
 
-    def __len__(self):
+    def __len__(self) -> int:
+        if self._total is None:
```

Comment:
isn't that gonna break somebodies workflow? 😄

Reply:
I don't think so. So what I was noticing is that sometimes __len__ gets called when it is None and thus a TypeError is raised when doing `int(None)`. But the caller (maybe `__iter__`) ignores it. But if I had an `assert self._total is not None` that blew up.

So it seems to be working but please let me know if you think something will break and can try to figure out a better way.

Reply:
So this is a sample program which shows how Python it ignores the type error, when using list()

```
$ cat test.py
import sys

class Testing:
    def __len__(self):
        print("Entered __len__")
        raise TypeError("we ignore this")
        return int(None)

    def __iter__(self):
        print("Entered __iter__")
        return self

    def __next__(self):
        raise StopIteration


def main() -> int:
    test = Testing()

    print(list(test))
    return 0


if "__main__" == __name__:
    sys.exit(main())
```

Run it:
```
$ python3 test.py
Entered __len__
Entered __iter__
Entered __len__
[]
```